### PR TITLE
Add tests for datetime.date

### PR DESCRIPTION
**Pull request by JonasNorlinder**
Add Date.java

Add datetime.today()

Implement Date attributes min, max and year

Implement replace()

Move range check of input to Date constructor

Add more tests to Date.replace()

Add documentation to Date.java and datetime.java

Check input of date()

Check input of replace()

Co-authored-by: Jonas Norlinder <jonas@norlinder.nu>

Run test with
`ant test && python setup.py test -s tests.stdlib.test_datetime.DatetimeModuleTests`

### CODE REVIEW

File: python/common/org/python/types/Date.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    public final static int MAX_MONTH = 11;
+    public final static int MIN_DAY = 1;
+    public final static int MAX_DAY = 31;
+
+    @org.python.Attribute()
+    public final static Date min = new Date(MIN_YEAR, MIN_MONTH, MIN_DAY);
+    @org.python.Attribute()
+    public final static Date max = new Date(MAX_YEAR, MAX_MONTH, MAX_DAY);
+    @org.python.Attribute()
+    public final static Date resolution = null; // TODO: implement `resolution` when timedelta is implemented
```
Comment (rehnarama) [inline python/common/org/python/types/Date.java]: Will you ignore this for now, or will you fix it after #34 is merged?

File: python/common/org/python/types/Date.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    @org.python.Attribute()
+    public final int day;
+
+    private final static java.text.SimpleDateFormat sdf = new java.text.SimpleDateFormat("yyyy-MM-dd");
+
+    public int hashCode() {
+        return this.value.hashCode();
+    }
+
+    public Date(int year, int month, int day) {
```
Comment (rehnarama) [inline python/common/org/python/types/Date.java]: Missing python type signature, see https://voc.readthedocs.io/en/latest/reference/signatures.html

File: python/common/python/datetime.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    @org.python.Attribute
+    public static org.python.Object __loader__ = org.python.types.NoneType.NONE; // TODO
+    @org.python.Attribute
+    public static org.python.Object __name__ = new org.python.types.Str("datetime");
+    @org.python.Attribute
+    public static org.python.Object __package__ = new org.python.types.Str("");
+    @org.python.Attribute
+    public static org.python.Object __spec__ = org.python.types.NoneType.NONE; // TODO
+
+    @org.python.Method(__doc__ = "date() -> new datetime object", default_args = { "year", "month", "day" })
```
Comment (rehnarama) [inline python/common/python/datetime.java]: Appropiate way would be to actually expose the constructor akin to how they have done it with OrderedDict (see https://github.com/Melkster/voc/blob/master/python/common/python/collections.java#L25-L51). 

What you have done here is simply a method which invokes the constructor for you, creating unneeded indirection

File: tests/stdlib/test_datetime.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
+from ..utils import TranspileTestCase
+from unittest import expectedFailure
+
+
+class DatetimeModuleTests(TranspileTestCase):
+    def test_date_constructor(self):
+        self.assertCodeExecution("""
+            import datetime
+            print(datetime.date(2000,2, 1))
```
Comment (rehnarama) [inline tests/stdlib/test_datetime.py]:
```suggestion
            print(datetime.date(2000, 2, 1))
```

File: tests/stdlib/test_datetime.py [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+            print(datetime.date(1,1,1))
```
Comment (rehnarama) [inline tests/stdlib/test_datetime.py]:
```suggestion
            print(datetime.date(1, 1, 1))
```

File: tests/stdlib/test_datetime.py [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+            print(datetime.date(9999,12,31))
```
Comment (rehnarama) [inline tests/stdlib/test_datetime.py]:
```suggestion
            print(datetime.date(9999, 12, 31))
```

**Review by rehnarama — CHANGES_REQUESTED** [review]
Some minor comments and style changes. Most important I'd say is to actually expose the constructor of your datetime object

File: python/common/org/python/types/Date.java (hunk 1 above)
Comment (mawkler) [reply python/common/org/python/types/Date.java]: I think the plan is to wait with it for now :)

**Comment by JonasNorlinder** [comment]
private org.python.Object getArg(int position, String keyword, org.python.Object[] args,
            java.util.Map<java.lang.String, org.python.Object> kwargs) is from pair 2

**Review by rehnarama — APPROVED** [review]
Looks good now 👍

**Pull request merged**